File: passenger-service/src/main/java/com/software/modsen/passengerservice/dto/PassengerCreateRequest.java
package com.software.modsen.passengerservice.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PassengerCreateRequest {
    @Pattern(regexp = "^[а-яА-Я]{2,}\\s?-?[а-яА-Я]{2,}$", message = "Incorrect firstname")
    @Schema(defaultValue = "firstname")
    private String firstname;

    @Pattern(regexp = "^[а-яА-Я]{2,}\\s?-?[а-яА-Я]{2,}$",message = "Incorrect surname")
    @Schema(defaultValue = "surname")
    private String surname;

    @Pattern(regexp = "^[а-яА-Я]{2,}\\s?-?[а-яА-Я]{2,}$",message = "Incorrect patronymic")
    @Schema(defaultValue = "patronymic")
    private String patronymic;

    @Email(message = "Incorrect email")
    @Schema(defaultValue = "email")
    private String email;

    @Pattern(regexp = "^\\+375(29|33|44|25)\\d{7}$", message = "Incorrect phone number")
    @Schema(defaultValue = "phone number")
    private String phoneNumber;
}

File: passenger-service/src/main/kotlin/com/software/modsen/passengerservice/dto/request/PassengerCreateRequest.kt
package com.software.modsen.passengerservice.dto.request

import com.software.modsen.passengerservice.util.CustomValidatePatterns
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Pattern

@Schema(description = "Data for create passenger")
data class PassengerCreateRequest (
    @field:Schema(defaultValue = "firstname")
    @field:Pattern(regexp = CustomValidatePatterns.NAME_PATTERN, message = "Incorrect firstname")
    val firstname: String,

    @field:Schema(defaultValue = "surname")
    @field:Pattern(regexp = CustomValidatePatterns.NAME_PATTERN, message = "Incorrect surname")
    val surname: String,

    @field:Schema(defaultValue = "patronymic")
    @field:Pattern(regexp = CustomValidatePatterns.NAME_PATTERN, message = "Incorrect patronymic")
    val patronymic: String,

    @field:Schema(defaultValue = "email")
    @field:Email(message = "Incorrect email")
    val email: String,

    @field:Schema(defaultValue = "phone number")
    @field:Pattern(regexp = CustomValidatePatterns.PHONE_NUMBER_PATTERN, message = "Incorrect phone number")
    val phoneNumber: String
)